managers/tests/test_views.py

```diff
@@ -258,119 +258,85 @@ def test_delete_project_function_view_should_not_delete_project_on_non_admin_req
-        other_manager = ManagerUserFactory()
-        self.client.force_login(other_manager)
+    def test_manage_task_activities_view_should_return_200_status_code_when_user_is_manager_or_admin(self):
+        manager = ManagerUserFactory()
+        admin = AdminUserFactory()
+        self.client.force_login(manager)
 
         response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 200)
 
-        self.assertEqual(response.status_code, 404)
-
-    def test_manage_task_activities_in_project_view_should_get_200_when_user_is_manager_of_current_project(self):
-        project_manager = ManagerUserFactory()
-        self.project.managers.add(project_manager)
-        self.client.force_login(project_manager)
-
+        self.client.force_login(admin)
         response = self.client.get(self.url)
-
         self.assertEqual(response.status_code, 200)
 
-    def test_manage_task_activities_in_project_view_should_redirect_on_employee_get(self):
+    def test_manage_task_activities_should_redirect_on_employee_get(self):
         user_employee = UserFactory()
-        self.project.members.add(user_employee)
-        self.client.force_login(user_employee)
 
+        self.client.force_login(user_employee)
         response = self.client.get(self.url)
 
         self.assertEqual(response.status_code, 302)
 
-    def test_manage_task_activities_view_should_show_all_task_activities_in_project_and_available_to_add(self):
-        self.assertCountEqual(self.project.project_activities.all(), TaskActivityType.objects.get_defaults())
-        self.assertNotIn(self.not_default_task_activity, self.project.project_activities.all())
+    def test_manage_task_activities_view_should_add_new_task_activity_on_post_and_it_should_be_active(self):
+        task_activity_name = "New activity"
+        response = self.client.post(self.url, data={"name": task_activity_name})
 
-        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, 302)
+        self.assertIn(
+            TaskActivityType.objects.get(name=task_activity_name), TaskActivityType.objects.filter(is_active=True)
+        )
 
-        self.assertEqual(response.status_code, 200)
-        self.assertContains(response, self.default_task_activity_1.name)
-        self.assertContains(response, self.not_default_task_activity.name)
+    def test_manage_task_activities_view_should_not_allow_to_create_two_tasks_with_the_same_name(self):
 
-    def test_manage_task_activities_view_should_add_new_task_activity_on_post_and_add_it_to_project(self):
         task_activity_name = "New activity"
-        response = self.client.post(self.url, data={"name": task_activity_name})
+        _number_of_task_activities_at_start = TaskActivityType.objects.all().count()
+        self.client.post(self.url, data={"name": task_activity_name})
+        self.client.post(self.url, data={"name": task_activity_name})
```

Comment:
You can ensure if this was done properly. Please change it to this:
```python
first_response = self.client.post(self.url, data={"name": task_activity_name}, follow=True)
self.assertEqual(first_response.status_code, 200)
self.assertNotIn("Task activity type with this Name already exists.", first_response.rendered_content)

second_response = self.client.post(self.url, data={"name": task_activity_name}, follow=True)
self.assertEqual(first_response.status_code, 200)
self.asserttIn("Task activity type with this Name already exists.", first_response.rendered_content)
```

Thanks to this will be sure that first time you don't have error message that something went wrong with duplicated task activity and in second you have. Below assert is okay and you can leave it as it is

Reply:
Changed